### DISCUSSION TITLE
White frame

### DISCUSSION THREAD

**Discussion by RiccardoFerrari7** [ImageMagick/ImageMagick#7176] [Help]
### ImageMagick version

7.0.8-59 Q16 x64

### Operating system, version and so on

win 10

### Description

How can I find the edges of an image?
X min, X max, Y min and Y max?
![Signed](https://github.com/ImageMagick/ImageMagick/assets/163754228/15c061c9-0844-4d76-a2df-74f67cdf6cd9)

**Comment by fmw42** [comment]
Please post a real image and show exactly what edges you want.

**Reply by RiccardoFerrari7** [reply]
![Original](https://github.com/ImageMagick/ImageMagick/assets/163754228/6b03e7b9-e7a7-42b8-82f0-fb60d5495e24)
Original image 4000 x 4000 
I need Max e Min not white pixels.
The red numbers in the previous image are the result I'm looking for.
Thank you for your attention.

**Reply by fmw42** [reply]
Sorry, I do not understand your question. Please post a real image and describe what rectangle you need.

**Reply by RiccardoFerrari7** [reply]
Thank you for your attention, but I don't understand why you don't get it.
I already posted the actual image. Obviously it's an example.
The four black lines are the image.
I need to find the highest black pixel, the lowest black pixel, the rightmost black pixel and the leftmost black pixel.
I have called them X max, X min, Y max and Y min.
In the example they are black but they could be any colour except white (in the example white = background colour).
I found the values by hand (the red one I already posted), I hope there is a way to do it automatically.

**Comment by dceeee** [comment]
I called your image whiteframe.png
magick whiteframe.png -format %@ info: 
reports 2959x2839+435+567

2959+435=3394
2839+567=3406

**Reply by RiccardoFerrari7** [reply]
Thanks, it seems to work. I'll do some tests tomorrow. So far it works fine in black and white, some problems with colour, but I'm probably doing something wrong.

**Reply by fmw42** [reply]
If the background is not pure white or some pure color, then add -fuzz XX% to your command to allow some tolerance. You can also try to convert to grayscale and threshold to make your image black/white before doing the %@.

**Reply by RiccardoFerrari7** [reply]
Thank you all.
It works.
